iothub/device/src/TelemetryMessage.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ComponentModel;
+using System.IO;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// A class used to send telmetry to IoT Hub.
+    /// </summary>
+    /// <remarks>
+    /// This class is derived from <see cref="Message"/> and is designed to accept a <see cref="TelemetryCollection"/>
+    /// to utilize the <see cref="PayloadConvention"/> to adhere to a well defined convention.
+    /// The convention defines the <see cref="ContentType"/> and <see cref="ContentEncoding"/> for this message.
+    /// </remarks>
+    public class TelemetryMessage : Message
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="TelemetryCollection"/> for this <see cref="TelemetryMessage"/>
+        /// </summary>
+        /// <value>A telemetry collection that will be set as the message payload.</value>
+        public TelemetryCollection Telemetry { get; set; } = new TelemetryCollection();
+
+        /// <inheritdoc/>
+        /// <remarks>
+        /// This property has been hidden by this class to only allow you to set the ContentType via the <see cref="PayloadSerializer"/> class.
+        /// </remarks>
+        /// <value>The base <see cref="Message"/> content type.</value>
+        [Browsable(false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public new string ContentType
```

Comment:
this is another point worth highlighting:
- since the content type and encoding info is now a part of the client itself (it is set in ClientOptions), these fields have no significance. Since we cannot hide them in the `Message` class directly (programatically), it has been hidden this way.

Thoughts?

Reply:
" these fields have no significance" => meaning there is no point in setting or getting this information from the `TelemetryMessage` class. It is set properly internally by the sdk though (in the `InternalClient`).

Reply:
One option would be to strip out everything but the ContentType and ContentEncoding from the Message class and make an abstract MessageBase that has the other properties we want to have. 

So it would be MessageBase -> Message and MessageBase -> TelemetryMessage.

We could then move that to the shared class and start the process of unifying the Message class

Reply:
This approach might help keep the Message class sealed as well.

Reply:
I agree, two birds one stone.